### ISSUE TITLE
sysctl doesn't work for keys that contain slash, e.g. VLANs

### ISSUE THREAD

**Issue by jktjkt** [voxpupuli/puppet-augeasproviders#102]
It is OK to have sysctl keys with slashes in them:

```
[root@potemkin02 ~]# sysctl net.ipv4.conf.bond0/122.rp_filter
net.ipv4.conf.bond0/122.rp_filter = 1
```

However, it seems that one cannot really apply them. The following Puppet manifest:

```
    sysctl { 'net.ipv4.conf.bond0/122.rp_filter':
        ensure  => present,
        value   => 2,
        require => Service['network'],
    }
```

...results in this:

<pre>Error: Could not set 'present' on ensure: /augeas/files/etc/sysctl.conf/error/path = /files/etc/sysctl.conf
/augeas/files/etc/sysctl.conf/error/lens = /usr/share/augeas/lenses/dist/sysctl.aug:36.10-.52:
/augeas/files/etc/sysctl.conf/error/message = Malformed child node 'net.ipv4.conf.bond0' at 105:/etc/puppet/environments/test_vlans/modules/vesnicky/manifests/network.pp
Error: Could not set 'present' on ensure: /augeas/files/etc/sysctl.conf/error/path = /files/etc/sysctl.conf
/augeas/files/etc/sysctl.conf/error/lens = /usr/share/augeas/lenses/dist/sysctl.aug:36.10-.52:
/augeas/files/etc/sysctl.conf/error/message = Malformed child node 'net.ipv4.conf.bond0' at 105:/etc/puppet/environments/test_vlans/modules/vesnicky/manifests/network.pp
Wrapped exception:
/augeas/files/etc/sysctl.conf/error/path = /files/etc/sysctl.conf
/augeas/files/etc/sysctl.conf/error/lens = /usr/share/augeas/lenses/dist/sysctl.aug:36.10-.52:
/augeas/files/etc/sysctl.conf/error/message = Malformed child node 'net.ipv4.conf.bond0'
Error: /Stage[main]/Vesnicky::Network/Sysctl[net.ipv4.conf.bond0/122.rp_filter]/ensure: change from absent to present failed: Could not set 'present' on ensure: /augeas/files/etc/sysctl.conf/error/path = /files/etc/sysctl.conf
/augeas/files/etc/sysctl.conf/error/lens = /usr/share/augeas/lenses/dist/sysctl.aug:36.10-.52:
/augeas/files/etc/sysctl.conf/error/message = Malformed child node 'net.ipv4.conf.bond0' at 105:/etc/puppet/environments/test_vlans/modules/vesnicky/manifests/network.pp</pre>

**Comment by trevor-vaughan** [comment]
I don't have an easy way to test this but why not munge the name so that all slashes are escaped in the provider?

**Comment by raphink** [comment]
@trevor-vaughan the problem though is down to the Augeas lens here. There is currently no way to set such keys in the lens.

**Comment by trevor-vaughan** [comment]
That makes sense and I certainly see the issue.

So, this would be a core patch to augeas to allow for the specification of an alternate key separator?

I would vote for a pipe since that tends to be the usual regex alternative that I've seen used.

**Comment by raphink** [comment]
@trevor-vaughan ah, you would just remap "/" as "|" in the labels?

**Comment by trevor-vaughan** [comment]
Yeah, I would give the user the ability to either use '/' or '|' as they desire so long as it is consistent within a given reference.

Alternatively, you could support escapes in keys but that seems hideous and prone to insanity.

**Comment by raphink** [comment]
As far as the Augeas lens is concerned, supporting `/` is not possible in labels, hence my suggestion to make a backward incompatible change.

**Comment by trevor-vaughan** [comment]
Would it be possible to support both? If '/' is the first character, that's the separator, if '|' is, then that is the separator.

Basically, either should be supported since that gives you a way to ensure that you cover all key cases (ok, except for keys that have both / and | but that's crazy and I don't want to live on those systems).

**Comment by raphink** [comment]
It's not possible in tree, but it could be remapped in the provider as you suggested. However, I'm not a big fan of this. `|` currently means nothing in sysctl, but who knows if it couldn't mean something in future versions? So I'd rather fix the lens properly (which imo means splitting on `\.|/`)

**Comment by raphink** [comment]
@trevor-vaughan when it comes to the lens itself, it would be better to discuss this in https://github.com/hercules-team/augeas/issues/176

**Comment by isaiahfrantz** [comment]
Has there been any fixes/workarounds for this issue?

**Comment by johnwarburton** [comment]
Please check https://github.com/voxpupuli/puppet-augeasproviders_sysctl/issues/17
